### ISSUE TITLE
In Godot 4.3 @export setters referencing nodes are called twice in nested scenes

### ISSUE THREAD

**Issue by graydoubt** [godotengine/godot#93094]
### Tested versions

- Reproducible in v4.3.beta1.official [a4f2ea91a]
- Reproducible in v4.3.dev6.official [89850d553]
- Reproducible in v4.3.dev5.official [89f70e98d]
- Reproducible in v4.3.dev4.official [df78c0636]
- Reproducible in v4.3.dev3.official [36e943b6b]
- Reproducible in v4.3.dev2.official [352434668]
- Reproducible in v4.3.dev1.official [9d1cbab1c]
- Not reproducible in v4.2.2.stable.official [15073afe3]

### System information

Godot v4.3.beta1 - Windows 10.0.19045 - Vulkan (Forward+) - dedicated NVIDIA GeForce GTX 1070 Ti (NVIDIA; 32.0.15.5599) - Intel(R) Core(TM) i7-4790K CPU @ 4.00GHz (8 Threads)

### Issue description

The setter for exported properties of type Node will be called twice if they're defined in a nested scene.

My `main` scene has a `ui` child scene.

![image](https://github.com/godotengine/godot/assets/36222624/077930a2-1e8f-4268-86d0-0be30772a2e3)

The `ui` child scene references a button:

![image](https://github.com/godotengine/godot/assets/36222624/774f9954-7921-4448-a07c-8217e3d21204)

With the script:

```gdscript
# In a nested scene
@export var my_button: Button:
	set(value):
		my_button = value
		print("%s: setting button" % [self])
```

If I run the `ui` scene directly (e.g. with F6), it will print (as expected):
```
UI:<CanvasLayer#27279754438>: setting button
```

If I run the `main` scene, it'll print it twice:

```
UI:<CanvasLayer#28571600128>: setting button
UI:<CanvasLayer#28571600128>: setting button
```

This started with 4.3-dev1.

### Steps to reproduce

Create a scene (e.g. called "ChildScene") and add an `@export`ed property that references a node (of any type) with a property setter.

Create another scene (e.g. called "ParentScene"), and add the `ChildScene` as a child scene.

Set a breakpoint in the property setter (or have it print out a debug message).

Run the `ParentScene` in Godot 4.3-dev1 or later, and see it invoke the breakpoint or print the message twice.

### Minimal reproduction project (MRP)

The MRP: https://github.com/graydoubt/godot-4.3-setter-called-twice

**Comment by graydoubt** [comment]
This may be related to how the `.tscn` files are stored in 4.3.

In Godot 4.2, the `main.tscn` would contain:

```
[gd_scene load_steps=2 format=3 uid="uid://d1bdwhmo0dn6f"]

[ext_resource type="PackedScene" uid="uid://bkucqgqglk6k1" path="res://ui.tscn" id="1_aniqm"]

[node name="Main" type="Node"]

[node name="UI" parent="." instance=ExtResource("1_aniqm")]
```

But in Godot 4.3, the `main` scene file redundantly stores the same properties already defined in the `ui` scene file:

```
[gd_scene load_steps=2 format=3 uid="uid://d1bdwhmo0dn6f"]

[ext_resource type="PackedScene" uid="uid://bkucqgqglk6k1" path="res://ui.tscn" id="1_aniqm"]

[node name="Main" type="Node"]

[node name="UI" parent="." node_paths=PackedStringArray("my_button", "my_label", "my_label2") instance=ExtResource("1_aniqm")]
my_button = NodePath("Button")
my_label = NodePath("Label")
my_label2 = NodePath("Label2")
```

It appears that this is why the setters are called twice. If the setter does any heavy work, it will perform it twice. If it connects signals, it'll try to connect them twice. This seems like a regression.

The full debug output from the MRP is:

```
<CanvasLayer#28571600128>: setting button path
<CanvasLayer#28571600128>: setting int
<CanvasLayer#28571600128>: setting string
<Button#28588377345>: setting icon
UI:<CanvasLayer#28571600128>: setting button
UI:<CanvasLayer#28571600128>: setting label
UI:<CanvasLayer#28571600128>: setting label2
UI:<CanvasLayer#28571600128>: setting button
UI:<CanvasLayer#28571600128>: setting label
UI:<CanvasLayer#28571600128>: setting label2
````

**Comment by KoBeWi** [comment]
Indeed, this is caused by default values being stored in the scene for Node properties. This was already reported and will be fixed in beta2: #91591
